weather_sp/splitter_pipeline/file_splitters.py

```diff
@@ -133,6 +135,40 @@ def _open_grib_locally(self) -> t.Iterator[t.Iterator[pygrib.gribmessage]]:
                 yield gb
 
 
+class GribSplitterV2(GribSplitter):
+    """Splitter that makes use of `grib_copy` util for high performance splitting.
+
+    See https://confluence.ecmwf.int/display/ECC/grib_copy.
+    """
+
+    def split_data(self) -> None:
+        if not self.output_info.split_dims():
+            raise ValueError('No splitting specified in template.')
+
+        if self.should_skip():
+            metrics.Metrics.counter('file_splitters', 'skipped').inc()
+            self.logger.info('Skipping %s, file already split.',
+                             repr(self.input_path))
+            return
+
+        output_template = self.output_info.unformatted_output_path().replace('{', '[').replace('}', ']')
+        root, output_template = os.path.split(output_template)
+        with self._copy_to_local_file() as local_file:
+            with tempfile.TemporaryDirectory() as tmpdir:
+                dest = os.path.join(tmpdir, output_template)
+
+                cmd = shutil.which('grib_copy')
+                if not cmd:
+                    raise EnvironmentError('binary `grib_copy` is not available in the current environment!')
```

Comment:
I think this check should already be performed earlier if possible.

Reply:
Great idea.